src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -123,6 +123,11 @@ internal static bool CanGetSemanticInfo(CSharpSyntaxNode node, bool allowNamedAr
             }
         }
 
+        /// <summary>
+        /// True if we should enable nullable analysis on information returned from this semantic model.
+        /// </summary>
+        internal bool EnableNullableAnalysis => Compilation.Options.NullableContextOptions != NullableContextOptions.Disable;
```

Comment:
>Compilation.Options.NullableContextOptions != NullableContextOptions.Disable [](start = 48, length = 76)

It feels like this is the wrong condition to check. We should simply check if the feature is enabled, i.e. check the language version.

Reply:
I'm leaving this as is for now, and we can resolve in the thread we have going on the subject.

---
In reply to: [269786876](https://github.com/dotnet/roslyn/pull/34429#discussion_r269786876) [](ancestors = 269786876)